src/write.jl

```diff
@@ -0,0 +1,40 @@
+function write(f::Function, path::AbstractString, ds::LazIO.LazDataset)
+    reader = ds.filehandle
+    # create writer
+    writer = Ref{Ptr{Cvoid}}(C_NULL)
+    LazIO.@check writer[] LazIO.laszip_create(writer)
+    # copy header from reader to writer (to be updated later)
+    LazIO.@check writer[] LazIO.laszip_set_header(writer[], Ref(ds.header))
+    # open writer
+    LazIO.@check writer[] LazIO.laszip_open_writer(writer[], path, Cint(1))
+    try
+        f(writer[])
+    finally
+        # close files and destroy pointers
+        LazIO.@check writer[] LazIO.laszip_close_writer(writer[])
+        LazIO.@check writer[] LazIO.laszip_destroy(writer[])
+    end
+end
+
+function write(f::Function, path::AbstractString, header::LazIO.laszip_header)
+    # create writer
+    writer = Ref{Ptr{Cvoid}}(C_NULL)
+    LazIO.@check writer[] LazIO.laszip_create(writer)
+    # copy header to writer (to be updated later)
+    LazIO.@check writer[] LazIO.laszip_set_header(writer[], Ref(header))
+    # open writer
+    LazIO.@check writer[] LazIO.laszip_open_writer(writer[], path, Cint(1))
+    try
+        f(writer[])
+    finally
+        # close files and destroy pointers
+        LazIO.@check writer[] LazIO.laszip_close_writer(writer[])
+        LazIO.@check writer[] LazIO.laszip_destroy(writer[])
+    end
+end
+
+function writepoint(writer::Ptr{Cvoid}, p::LazIO.laszip_point)
```

Comment:
Why pass it as pointer here? Before everything is used as Reference. It makes the decision of using [] even more vague.

Reply:
If we want a Ref here we need to use `f(writer)` in the `write` function, which is possible. We probably need to think about the Ref situation a bit more and check for consistency. Feel free to change this one already.